parlai/tasks/ubuntu/agents.py

```diff
@@ -46,3 +46,74 @@ def setup_data(self, path):
                     cands.append(response)
                     random.shuffle(cands)
                 yield (context, [response], None, cands), True
+
+
+class MultiTeacher(FixedDialogTeacher):
+    def __init__(self, opt, shared=None):
+        self.datatype = opt['datatype']
+        opt['datafile'] = os.path.join(
+            opt['datapath'], 'Ubuntu', opt['datatype'].split(':')[0] + '.csv'
+        )
+        super().__init__(opt, shared)
+
+        self.opt = opt
```

Comment:
i believe this is done in the call to `super().__init__`, so you can drop this line.